src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,18 +40,47 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+
+
     }
 
     /**
      * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+        ArrayList<String> separatedFormat = splitByComma(test);
+        if(separatedFormat.size() == 4) {
+            block = new Block(separatedFormat.get(0).trim());
+            street = new Street(separatedFormat.get(1).trim());
+            unit = new Unit(separatedFormat.get(2).trim());
+            postal = new Postal(separatedFormat.get(3).trim());
+            return true;
```

Comment:
What would happen if an invalid value was supplied for block/street etc? May want to use regex to ensure address validation is thorough.